xbmc/interfaces/json-rpc/PVROperations.cpp

```diff
@@ -145,6 +146,40 @@ JSONRPC_STATUS CPVROperations::GetChannelDetails(const std::string &method, ITra
   return OK;
 }
 
+JSONRPC_STATUS CPVROperations::GetClients(const std::string& method,
+                                          ITransportLayer* transport,
+                                          IClient* client,
+                                          const CVariant& parameterObject,
+                                          CVariant& result)
+{
+  if (!CServiceBroker::GetPVRManager().IsStarted())
+    return FailedToExecute;
+
+  int start, end;
+
+  auto clientInfos = CServiceBroker::GetPVRManager().Clients()->GetEnabledClientInfos();
+  HandleLimits(parameterObject, result, clientInfos.size(), start, end);
+  for (int index = start; index < end; index++)
+  {
+    CVariant object(CVariant::VariantTypeObject);
+    object["clientid"] = clientInfos[index].m_clientId;
+    object["addonid"] = clientInfos[index].m_addonId;
+    object["label"] = clientInfos[index].m_clientName;
+
+    object["supportstv"] = clientInfos[index].m_supportsTV;
+    object["supportsradio"] = clientInfos[index].m_supportsRadio;
+    object["supportsepg"] = clientInfos[index].m_supportsEPG;
+    object["supportsrecordings"] = clientInfos[index].m_supportsRecordings;
+    object["supportstimers"] = clientInfos[index].m_supportsTimers;
+    object["supportschannelgroups"] = clientInfos[index].m_supportsChannelGroups;
+    object["supportschannelscan"] = clientInfos[index].m_supportsChannelScan;
```

Comment:
Not sure this should be implemented here. The other PVR classes implement ISerializable and in Serialize() happens what you've coded here. Don't know whether this matters, just seeing the difference.

Another point is that implementing ISerializable would make the clientinfo struct obsolete, which you've introduced just to get client data here.

Reply:
It's a fair point. I modelled it on how channel groups were done as these are not wrapped in file items either.

The code that generates these is based on file item lists and file items which these would not be.

Reply:
Have switched this to pass variant list instead of the new struct. Much simpler, thanks for the guidance. So latest should not finally be good to go.